client/login/magic-login/gravatar/index.tsx

```diff
@@ -0,0 +1,1287 @@
+		return () => {
+			if ( resendEmailCountdownId.current ) {
+				clearInterval( resendEmailCountdownId.current );
+				resendEmailCountdownId.current = null;
+			}
+		};
+	}, [] );
+
+	const handleGravPoweredEmailCodeSend = async ( email: string, cb = () => {} ) => {
+		const noticeId = 'email-code-notice';
+		const duration = 4000;
+		const eventOptions = { client_id: oauth2ClientId, client_name: oauth2Client?.title };
+
+		setIsRequestingEmail( true );
+
+		try {
+			infoNotice( translate( 'Sending email…' ), { id: noticeId, duration } );
+
+			recordTracksEvent(
+				'calypso_gravatar_powered_magic_login_email_code_requesting',
+				eventOptions
+			);
+
+			const { public_token } = await wpcom.req.post(
+				'/auth/send-login-email',
+				{ apiVersion: '1.3' },
+				{
+					client_id: config( 'wpcom_signup_id' ),
+					client_secret: config( 'wpcom_signup_key' ),
+					locale,
+					lang_id: getLanguage( locale )?.value,
+					email: email,
+					redirect_to: currentQueryArguments?.email_address,
+					flow: oauth2Client ? getGravatarOAuth2Flow( oauth2Client ) : undefined,
+					create_account: true,
+					tos: getToSAcceptancePayload(),
+					token_type: 'code',
+					...( isSecondaryEmail ? { gravatar_main: ! isNewAccount } : {} ),
+				}
+			);
+
+			setPublicToken( public_token );
+			setShowEmailCodeVerification( true );
+			startResendEmailCountdown();
+			cb();
+
+			dispatch( removeNotice( noticeId ) );
+			dispatch( successNotice( translate( 'Email Sent. Check your mail app!' ), { duration } ) );
+
+			recordTracksEvent( 'calypso_gravatar_powered_magic_login_email_code_success', eventOptions );
+		} catch ( error ) {
+			const err = error as { error?: string; status?: number; message?: string };
+			if ( err?.error ) {
+				setRequestEmailErrorMessage( err?.message ?? null );
+			} else {
+				setRequestEmailErrorMessage( translate( 'Something went wrong. Please try again.' ) );
+			}
+
+			dispatch( removeNotice( noticeId ) );
+			dispatch( errorNotice( translate( 'Sorry, we couldn’t send the email.' ), { duration } ) );
+
+			recordTracksEvent( 'calypso_gravatar_powered_magic_login_email_code_failure', {
+				...eventOptions,
+				error_code: err?.status,
+				error_message: err?.message,
+			} );
+		}
+
+		setIsRequestingEmail( false );
+	};
+
+	const handleGravPoweredEmailSwitch = () => {
+		setShowSecondaryEmailOptions( false );
+		setShowEmailCodeVerification( false );
+		setVerificationCodeInputValue( '' );
+		setIsNewAccount( false );
+
+		dispatch( hideMagicLoginRequestForm() );
+
+		recordTracksEvent( 'calypso_gravatar_powered_magic_login_click_switch_email', {
+			client_id: oauth2ClientId,
+			client_name: oauth2Client?.title,
+		} );
+	};
+
+	const handleGravPoweredCodeSubmit = ( e: React.FormEvent< HTMLFormElement > ) => {
+		e.preventDefault();
+
+		dispatch(
+			fetchMagicLoginAuthenticate(
+				`${ publicToken }:${ btoa( verificationCodeInputValue ) }`,
+				currentQueryArguments?.email_address ?? '',
```

Comment:
@chriskmnds Same as [this one](https://github.com/Automattic/wp-calypso/pull/104863#discussion_r2267613127).

```suggestion
				currentQueryArguments?.redirect_to ?? '',
```

Reply:
Weird. I was certain I fixed this before 😆